client/x/evmengine/keeper/msg_server.go

```diff
@@ -124,9 +127,7 @@ func (s msgServer) ExecutionPayload(ctx context.Context, msg *types.MsgExecution
 
 			return false, nil // Retry
 		} else if isSyncing(fcr.PayloadStatus) {
-			log.Warn(ctx, "Processing finalized payload halted while evm syncing (will retry)", nil, "payload_height", payload.Number)
-
-			return false, nil // Retry
```

Comment:
Does snap sync require `return false, nil // Retry` line to be removed here?

Asking since this changes the expected behavior. Originally, we expected `isSyncing == true` to retry the fetch logic. But removing `return false, nil` means `retryForever` will stop retrying when the node is syncing (or accepted) and continue to the next logic of delivering EL events to CL handlers.

I believe this will lead to non-determinism for a node if its execution client keeps returning `SYNCING` or `ACCEPTED` and eventually exceed the expected block time and progresses to the next block with non-updated execution client (which shouldn't happen in the original behavior, where the node should hang as it awaits syncing).

Reply:
Yes it is required for snap sync by requesting `forkChoiceUpdated` with the latest block.

The `forkChoiceUpdated` here is just to set safe and finalized block of EL chains. No fetched data is not required and used for finalized CL block. Only the data from finalized msg is used to process `FinalizeBlock`. That means, in CL perspective, it is fine to process the block with the finalized msg (just EL is syncing), not leading to non-determinism.

As I [mentioned](https://github.com/piplabs/trust-story/issues/43#issuecomment-2727804187) in the trust issue, the non-updated payload is not imported immediately. It is stashed for the update later by `forkChoiceUpdated`.